core/state_transition.go

```diff
@@ -270,8 +283,18 @@ func (st *StateTransition) payFees(eHardFork bool) error {
 		feeVal.Add(feeVal, st.msg.GatewayFee())
 	}
 
+	// Check if balance < GasFeeCap * gas + value
+	balanceCheck := feeVal
+	if st.gasFeeCap != nil {
```

Comment:
The balance check should be rolled into `canPayFee`

Reply:
Which first/second tests are you referring to?

Reply:
the `have, want` check inside this function & then the `canPayFees` function are the two checks against a too low balance